lib/runners/processor.rb

```diff
@@ -11,28 +11,16 @@ def self.register_config_schema(**args)
       Schema::Config.register(**args)
     end
 
-    def initialize(guid:, workspace:, git_ssh_path:, trace_writer:)
+    def initialize(guid:, workspace:, config:, git_ssh_path:, trace_writer:)
       @guid = guid
       @workspace = workspace
       @working_dir = workspace.working_dir
       @git_ssh_path = git_ssh_path
       @trace_writer = trace_writer
       @warnings = []
-      @ci_config =
-        if ci_config_path.file?
-          begin
-            YAML.load_file(ci_config_path.to_s, fallback: {}).tap do |conf|
-              trace_writer.ci_config(conf, file: ci_config_path.basename.to_s)
-            end
-          rescue Psych::SyntaxError => exn
-            message = "Your `#{relative_path(ci_config_path.to_s)}` file may be broken (line #{exn.line}, column #{exn.column})."
-            trace_writer.error message
-            raise CIConfigBroken.new(message)
-          end
-        end
-      # Runner maybe break `@ci_config`.
-      # So we should deep copy it to avoid collecting incorrect config file.
-      @ci_config_for_collect = Marshal.load(Marshal.dump(@ci_config))
+      @config = config.tap do |c|
+        c.content&.tap { |content| trace_writer.ci_config(content, file: c.path_name) }
+      end
```

Comment:
[suggestion] What about this refactoring to clarify the code intent?

```suggestion
      @config = config
      trace_writer.ci_config(config.content, file: config.path_name) if config.path_exist?
```

Reply:
Thank you. Done with 00ba624